Gems/AtomLyIntegration/CommonFeatures/Code/Source/Mesh/MeshComponentController.cpp

```diff
@@ -699,6 +701,81 @@ namespace AZ
+                const AZ::Data::Asset<AZ::RPI::BufferAsset> indexBufferAsset = indexBufferView.GetBufferAsset();
+                if (!indexBufferAsset.IsReady() || indexBufferViewDesc.m_elementSize != sizeof(uint32_t))
+                {
+                    AZ_Warning("MeshComponentController", false, "Unable to get geometry for mesh because index buffer asset is not ready or is an incompatible format.");
+                    continue;
+                }
+
+                // Get the position buffer data, if it exists with the expected name.
+                const AZ::RPI::BufferAssetView* positionBufferView = mesh.GetSemanticBufferAssetView(positionName);
+                if (!positionBufferView)
+                {
+                    AZ_Warning("MeshComponentController", false, "Unable to get geometry for mesh because position buffer data was not found.");
+                    continue;
+                }
+
+                // Confirm that the position buffer is valid and contains 3 32 bit floats for each position. Other formats are currently not
+                // supported.
+                constexpr uint32_t ElementsPerVertex = 3;
+                const AZ::RHI::BufferViewDescriptor& positionBufferViewDesc = positionBufferView->GetBufferViewDescriptor();
+                const AZ::Data::Asset<AZ::RPI::BufferAsset> positionBufferAsset = positionBufferView->GetBufferAsset();
+                if (!positionBufferAsset.IsReady() || positionBufferViewDesc.m_elementSize != sizeof(float) * ElementsPerVertex)
+                {
+                    AZ_Warning("MeshComponentController", false, "Unable to get geometry for mesh because position buffer asset is not ready or is an incompatible format.");
+                    continue;
+                }
+
+                const AZStd::span<const uint8_t> indexRawBuffer = indexBufferAsset->GetBuffer();
+                const uint32_t* indexPtr = reinterpret_cast<const uint32_t*>(
+                    indexRawBuffer.data() + (indexBufferViewDesc.m_elementOffset * indexBufferViewDesc.m_elementSize));
+
+                const AZStd::span<const uint8_t> positionRawBuffer = positionBufferAsset->GetBuffer();
+                const float* positionPtr = reinterpret_cast<const float*>(
+                    positionRawBuffer.data() + (positionBufferViewDesc.m_elementOffset * positionBufferViewDesc.m_elementSize));
+
+                // Copy the index and position data into the visible geometry structure.
+                AzFramework::VisibleGeometry visibleGeometry;
+                visibleGeometry.m_transform = AZ::Matrix4x4::CreateFromTransform(m_transformInterface->GetWorldTM());
+
+                // This resize and copy assumes the stride between elements is 0.
+                visibleGeometry.m_indices.resize_no_construct(indexBufferViewDesc.m_elementCount);
+                memcpy(&visibleGeometry.m_indices[0], indexPtr, indexBufferViewDesc.m_elementCount * indexBufferViewDesc.m_elementSize);
```

Comment:
These memcopies assume int/float vectors. Maybe add validation like this
AZ_Assert( (sizeof(visibleGeometry.m_indices[0]) * visibleGeometry.m_indices.size())>= indexBufferViewDesc.m_elementCount * indexBufferViewDesc.m_elementSize, "Error message");